### PR TITLE
[Feature] Our Courses

### PR DESCRIPTION
**Pull request by gertie-sheshe**
## Description

Add Our Courses section.

## Type of change

Please delete options that are not relevant.

- [ ] New feature (non-breaking change which adds functionality)

## Screenshots
Screenshot 2021-08-10 at 22 47 32
Screenshot 2021-08-10 at 22 47 45
Screenshot 2021-08-10 at 22 47 59

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

**Comment by kilemensi** [comment]
Isn't this PR a duplicate of https://github.com/CodeForAfrica/PesaYetu/pull/54 @gertie-sheshe ?

**Comment by kilemensi** [comment]
QQs @gertie-sheshe 

1. Is there **_any_** difference between the card in this PR and `InsightCard` used on homepage to show data-driven insights?
2. Where is the CMS integration?

File: src/components/OurCourseCard/index.stories.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+import React from "react";
+
+import OurCourseCard from "@/pesayetu/components/OurCourseCard";
+import { ourCourses } from "@/pesayetu/config";
+
+const { items } = ourCourses;
+const [dataItem] = items;
+
+export default {
+  title: "Components/Course ard",
```
Comment (KhadijaMahanga) [inline src/components/OurCourseCard/index.stories.js:11]: ` `

File: src/components/OurCourses/index.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+import OurCourseCard from "@/pesayetu/components/OurCourseCard";
+import Section from "@/pesayetu/components/Section";
+
+const OurCourses = ({ title, items, ...props }) => {
+  let itemsData = [];
+  const theme = useTheme();
+  const isTablet = useMediaQuery(theme.breakpoints.only("md"));
+
+  itemsData = isTablet ? items.slice(0, 2) : items;
```
Comment (KhadijaMahanga) [inline src/components/OurCourses/index.js:16]: Missing carousel on mobile design.. and that should apply on tablets too.

**Comment by gertie-sheshe** [comment]
> Isn't this PR a duplicate of #54 @gertie-sheshe ?

Yeah, I have no idea why two were created, I only recall creating one. The other is closed though so shouldn't be a problem

File: src/components/OurCourseCard/useStyles.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
+import { makeStyles } from "@material-ui/core/styles";
+
+const useStyles = makeStyles(({ typography, breakpoints }) => ({
+  root: {},
```
[the remark this answers is not in the document]
Comment (gertie-sheshe) [reply src/components/OurCourseCard/useStyles.js:4]: You mean [this](https://github.com/CodeForAfrica/PesaYetu/pull/55/files#diff-25c04fdeca7537f67cfd93ce134e40d0968884d258cf03f55947cc72d5b39176R5)? @KhadijaMahanga

File: src/components/OurCourses/index.stories.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import React from "react";
+
+import OurCourseCard from "@/pesayetu/components/OurCourseCard";
+import { ourCourses } from "@/pesayetu/config";
+
+export default {
+  title: "Sections/InsightsData",
```
[the remark this answers is not in the document]
Comment (gertie-sheshe) [reply src/components/OurCourses/index.stories.js]: Fixed

File: src/components/OurCourses/index.js [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+      <Section classes={{ root: classes.section }}>
+        {title && (
+          <Typography variant="h3" className={classes.title}>
+            {title}
+          </Typography>
+        )}
+
+        <Grid container className={classes.list}>
+          {itemsData?.map((item) => {
+            return (
```
[the remark this answers is not in the document]
Comment (gertie-sheshe) [reply src/components/OurCourses/index.js]: Fixed

File: src/components/OurCourseCard/index.stories.js (hunk 1 above)
Comment (gertie-sheshe) [reply src/components/OurCourseCard/index.stories.js]: Fixed

File: src/components/OurCourseCard/useStyles.js [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+  card: {
+    maxWidth: typography.pxToRem(350),
```
[the remark this answers is not in the document]
Comment (gertie-sheshe) [reply src/components/OurCourseCard/useStyles.js:6]: Fixed

File: src/components/OurCourses/index.js (hunk 2 above)
Comment (gertie-sheshe) [reply src/components/OurCourses/index.js]: Done

File: src/components/OurCourses/index.js [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+              <Grid item lg={4} xs={12} md={6}>
+                <OurCourseCard key={item.title} {...item} />
```
[the remark this answers is not in the document]
Comment (gertie-sheshe) [reply src/components/OurCourses/index.js]: Fixed

File: src/components/OurCourseCard/useStyles.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (KhadijaMahanga) [reply src/components/OurCourseCard/useStyles.js:4]: @gertie-sheshe yes

File: src/pages/index.js [hunk 8]
```diff
@@ -23,6 +24,7 @@ export default function Home({ boundary, blocks, ...props }) {
       <Hero {...heroProps} boundary={boundary} />
       <HowItWorks {...blocks?.howItWorks} />
       <DataVisuals {...blocks?.dataVisuals} />
+      <OurCourses {...blocks?.ourCourses} />
```
Comment (kilemensi) [inline src/pages/index.js:27]: Our Courses doesn't belong on the homepage.

File: src/components/OurCourses/index.js [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+        </Hidden>
+
+        <Hidden lgUp>
+          <Carousel
+            swipeable
+            responsive={responsive}
+            arrows={false}
+            renderDotsOutside
+            showDots
+            dotListClass={classes.dots}
```
Comment (kilemensi) [inline src/components/OurCourses/index.js:50]: `No need to separate `mdDown` and lgUp`... just set `showDots={false}` in desktop mode and we're done.

**Review by kilemensi — COMMENTED** [review]
👍🏾 

----

You're still creating a whole new Card component, `OurCourseCard`, without stating what is the difference between `OurCourseCard` and `[InsightCard](https://github.com/CodeForAfrica/PesaYetu/tree/main/src/components/InsightCard)`.

File: src/pages/index.js (hunk 8 above)
Comment (gertie-sheshe) [reply src/pages/index.js]: Fixed

File: src/components/OurCourses/index.js (hunk 9 above)
Comment (gertie-sheshe) [reply src/components/OurCourses/index.js]: Fixed

File: src/components/OurCourses/index.js [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+        {title && (
+          <Typography variant="h3" className={classes.title}>
+            {title}
+          </Typography>
+        )}
+        <Hidden mdDown>
+          <Grid container className={classes.list}>
+            {items?.map((item) => (
+              <Grid item lg={4} key={item.title}>
+                <OurCourseCard {...item} />
```
[the remark this answers is not in the document]
Comment (gertie-sheshe) [reply src/components/OurCourses/index.js]: Fixed

File: src/components/OurCourses/index.js [hunk 11]
```diff
@@ -0,0 +1,79 @@
...
+          swipeable
+          responsive={responsive}
+          arrows={false}
+          renderDotsOutside
+          showDots={!isDesktop}
+          dotListClass={classes.dots}
+        >
+          {items?.map((item) => (
+            <Grid item key={item.title}>
+              <OurCourseCard {...item} />
```
Comment (kilemensi) [inline src/components/OurCourses/index.js:54]: Just like you did for AboutTeam @gertie-sheshe? Why aren't we just importing `InsightCard` and use it here?

Comment (gertie-sheshe) [reply src/components/OurCourses/index.js:54]: It is the InsightCard, I just changed the import name 😅 @kilemensi

Comment (kilemensi) [reply src/components/OurCourses/index.js:54]: Cool @gertie-sheshe ... then why is [`OurCourseCard`](https://github.com/CodeForAfrica/PesaYetu/pull/55/files#diff-47c6b9a0c492fa3047847e3a077bdda3156a05a53712ae1e733c4b7b04cb59c6R1) still around?

Comment (gertie-sheshe) [reply src/components/OurCourses/index.js:54]: ooh..yeah, you're right. Fixed

File: src/components/OurCourses/useStyles.js [hunk 12]
```diff
@@ -0,0 +1,51 @@
...
+    paddingLeft: typography.pxToRem(20),
+    paddingRight: typography.pxToRem(20),
+    [breakpoints.up("md")]: {
+      padding: `${typography.pxToRem(80)} 0`,
+    },
+  },
+  title: {
+    fontWeight: "900",
+    fontSize: typography.pxToRem(20),
+    marginBottom: typography.pxToRem(40),
```
Comment (kilemensi) [inline src/components/OurCourses/useStyles.js:18]: This is already defined in [theme](https://github.com/CodeForAfrica/PesaYetu/blob/08f271db28c6417e188bfb0c0ebd7375684465e7/src/theme/index.js#L141-L145) @gertie-sheshe

File: src/components/OurCourses/index.js [hunk 13]
```diff
@@ -0,0 +1,79 @@
...
+        <Carousel
+          swipeable
+          responsive={responsive}
+          arrows={false}
+          renderDotsOutside
+          showDots={!isDesktop}
+          dotListClass={classes.dots}
+        >
+          {items?.map((item) => (
+            <Grid item key={item.title}>
```
Comment (kilemensi) [inline src/components/OurCourses/index.js:53]: Why `Grid item`?

File: src/components/OurCourses/useStyles.js [hunk 14]
```diff
@@ -0,0 +1,51 @@
+import { makeStyles } from "@material-ui/core/styles";
+
+const useStyles = makeStyles(({ typography, breakpoints }) => ({
+  root: {
+    backgroundColor: "#DFDFDF",
```
Comment (kilemensi) [inline src/components/OurCourses/useStyles.js:5]: Already defined in [theme](https://github.com/CodeForAfrica/PesaYetu/blob/08f271db28c6417e188bfb0c0ebd7375684465e7/src/theme/index.js#L44)

**Review by kilemensi — COMMENTED** [review]
👍🏾

File: src/components/OurCourses/useStyles.js (hunk 12 above)
Comment (gertie-sheshe) [reply src/components/OurCourses/useStyles.js]: Fixed

File: src/components/OurCourses/useStyles.js (hunk 14 above)
Comment (gertie-sheshe) [reply src/components/OurCourses/useStyles.js]: Fixed

File: src/components/OurCourses/index.js (hunk 13 above)
Comment (gertie-sheshe) [reply src/components/OurCourses/index.js]: Fixed

File: src/components/OurCourses/useStyles.js [hunk 15]
```diff
@@ -0,0 +1,49 @@
...
+  section: {
+    paddingTop: typography.pxToRem(60),
+    paddingLeft: typography.pxToRem(20),
+    paddingRight: typography.pxToRem(20),
+    [breakpoints.up("md")]: {
+      padding: `${typography.pxToRem(80)} 0`,
+    },
+  },
+  title: {
+    fontSize: typography.pxToRem(20),
```
Comment (kilemensi) [inline src/components/OurCourses/useStyles.js:16]:
```suggestion
```
https://github.com/CodeForAfrica/PesaYetu/blob/179b9bf1da57f79ac36ca2fcac178362c6525508/src/theme/index.js#L155-L158

File: src/components/OurCourses/useStyles.js [hunk 16]
```diff
@@ -0,0 +1,49 @@
...
+      paddingBottom: typography.pxToRem(0),
+    },
+  },
+
+  list: {
+    flexDirection: "column",
+    [breakpoints.up("md")]: {
+      flexDirection: "row",
+    },
+  },
```
Comment (kilemensi) [inline src/components/OurCourses/useStyles.js:46]:
```suggestion
```

Or is it being used anywhere?

**Review by kilemensi — APPROVED** [review]
🚀 

---

Courses should link to https://courses.academy.africa/ for now (pick any courses there).

**Review by KobbyMmo — APPROVED** [review]
Courses do not have a hover state, so we should take the background on hover out.

image

File: src/components/OurCourses/index.js [hunk 17]
```diff
@@ -0,0 +1,79 @@
...
+          responsive={responsive}
+          arrows={false}
+          renderDotsOutside
+          showDots={!isDesktop}
+          dotListClass={classes.dots}
+        >
+          {items?.map((item) => (
+            <div key={item.title}>
+              <OurCourseCard {...item} />
+            </div>
```
[the remark this answers is not in the document]
Comment (gertie-sheshe) [reply src/components/OurCourses/index.js]: Works

**Review by BrendaBucha — APPROVED** [review]
@gertie-sheshe 🚀 For the hover effect for the card, I had a fix applied here https://github.com/CodeForAfrica/PesaYetu/blob/1f02ddee55fd1b4b9b3cf73db6fd78283e0da8dc/src/components/Card/index.js#L26

It should work for all components using the card component

**Review by KhadijaMahanga — APPROVED** [review]

**Pull request merged**